erpnext/accounts/doctype/journal_entry_template/journal_entry_template.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.model.document import Document
+from erpnext.accounts.utils import get_balance_on
+
+class JournalEntryTemplate(Document):
+	def autoname(self):
+		self.name = self.voucher_type + ' - ' + frappe.get_value('Company', self.company, 'abbr')
```

Comment:
Add a field 'Template Title' which should be used for naming, autoname not required.